### PR TITLE
Backfill tests for tracing packages

### PR DESCRIPTION
**Pull request by RebeccaMahany**
Adds tests for the exporter package; makes some small updates and bugfixes.

### CODE REVIEW

File: pkg/traces/exporter/client_authenticator.go [hunk 1]
```diff
@@ -22,6 +24,12 @@ func (c *clientAuthenticator) setToken(token string) {
 	c.token = token
 }
 
+// setDisableTLS updates the return value of RequireTransportSecurity. TLS should
+// be disabled only for local testing.
+func (c *clientAuthenticator) setDisableTLS(disableTLS bool) {
+	c.disableTLS = disableTLS
+}
```
Comment (directionless) [inline pkg/traces/exporter/client_authenticator.go:31]: If we only need this for tests, we can define this in the `_test.go` file.

File: pkg/traces/exporter/client_authenticator.go [hunk 2]
```diff
@@ -7,12 +7,14 @@ import (
 
 // Implements google.golang.org/grpc/credentials.PerRPCCredentials interface
 type clientAuthenticator struct {
-	token string
+	token      string
+	disableTLS bool
 }
 
-func newClientAuthenticator(token string) *clientAuthenticator {
+func newClientAuthenticator(token string, disableTLS bool) *clientAuthenticator {
```
Comment (directionless) [inline pkg/traces/exporter/client_authenticator.go:14]: With `setDisableTLS` do you need to change the signature here? It feels weird to change it for something only in CI

File: pkg/traces/exporter/client_authenticator.go (hunk 1 above)
Comment (RebeccaMahany) [reply pkg/traces/exporter/client_authenticator.go:31]: I actually don't want it only for tests! `RequireTransportSecurity` is part of the `google.golang.org/grpc/credentials.PerRPCCredentials` interface and I'd been just hardcoding the return to false for initial testing. But now that we've got a proper flag for `disable_ingest_tls` I wanted to plumb that through here as well.

Comment (RebeccaMahany) [reply pkg/traces/exporter/client_authenticator.go:31]: (By "local testing" in the comment I mean that we would only set `disable_ingest_tls` to false when testing against e.g. a local collector running in docker.)

Comment (directionless) [reply pkg/traces/exporter/client_authenticator.go:31]: okay

**Review by directionless — APPROVED** [review]

**Pull request merged**